core/src/main/java/org/apache/iceberg/rest/RESTJacksonModule.java

```diff
@@ -0,0 +1,183 @@
+        .addSerializer(SortOrder.class, new SortOrderSerializer())
+        .addDeserializer(SortOrder.class, new SortOrderDeserializer());
+    mapper.registerModule(module);
+
+    mapper.setVisibility(PropertyAccessor.ALL, JsonAutoDetect.Visibility.NONE);
+    mapper.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY);
+  }
+
+  public static class NamespaceDeserializer extends JsonDeserializer<Namespace> {
+    @Override
+    public Namespace deserialize(JsonParser p, DeserializationContext ctxt) throws IOException {
+      JsonNode jsonNode = p.getCodec().readTree(p);
+      return namespaceFromNode(p, jsonNode);
+    }
+  }
+
+  public static class NamespaceSerializer extends JsonSerializer<Namespace> {
+    @Override
+    public void serialize(Namespace namespace, JsonGenerator gen, SerializerProvider serializers)
+        throws IOException {
+      String[] parts = namespace.levels();
+      gen.writeArray(parts, 0, parts.length);
+    }
+  }
+
+  public static class TableIdentifierDeserializer extends JsonDeserializer<TableIdentifier> {
+    @Override
+    public TableIdentifier deserialize(JsonParser p, DeserializationContext ctxt)
+        throws IOException {
+      JsonNode jsonNode = p.getCodec().readTree(p);
+      if (!jsonNode.isObject()) {
+        throw new JsonParseException(p, "TableIdentifier must be an object: " + jsonNode);
+      }
+      ObjectNode objectNode = (ObjectNode) jsonNode;
+      Namespace namespace = namespaceFromNode(p, objectNode.get("namespace"));
+      String name = objectNode.get("name").asText();
+      return TableIdentifier.of(namespace, name);
+    }
+  }
+
+  private static Namespace namespaceFromNode(JsonParser parser, JsonNode jsonNode) throws IOException {
+    if (!jsonNode.isArray()) {
+      throw new JsonParseException(parser, "Namespace must be an array: " + jsonNode);
+    }
+    ArrayNode arrayNode = (ArrayNode) jsonNode;
+    String[] levels = new String[arrayNode.size()];
+    for (int i = 0; i < levels.length; i++) {
+      levels[i] = arrayNode.get(i).asText();
+    }
+    return Namespace.of(levels);
+  }
+
+  public static class TableIdentifierSerializer extends JsonSerializer<TableIdentifier> {
+    @Override
+    public void serialize(
+        TableIdentifier identifier, JsonGenerator gen, SerializerProvider serializers)
+        throws IOException {
+      gen.writeStartObject();
+      gen.writeFieldName("namespace");
+      String[] levels = identifier.namespace().levels();
+      gen.writeArray(levels, 0, levels.length);
+      gen.writeStringField("name", identifier.name());
+      gen.writeEndObject();
+    }
+  }
+
+  public static class SchemaDeserializer extends JsonDeserializer<Schema> {
+    @Override
+    public Schema deserialize(JsonParser p, DeserializationContext context) throws IOException {
+      JsonNode jsonNode = p.getCodec().readTree(p);
+      Schema schema = SchemaParser.fromJson(jsonNode);
+      context.setAttribute("schema", schema);
```

Comment:
What is the scope of the context that this is modifying?

Does the schema need to be deserialized before the partition spec?